src/main/java/core/basesyntax/ShopServiceImpl.java

```diff
@@ -0,0 +1,19 @@
+package core.basesyntax;
+
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private final OperationStrategy operationStrategy;
+
+    public ShopServiceImpl(OperationStrategy operationStrategy) {
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void process(List<FruitTransaction> transactions, Storage storage) {
+        for (FruitTransaction transaction : transactions) {
+            OperationHandler handler = operationStrategy.getHandler(transaction.getOperation());
+            handler.handle(transaction, storage);
+        }
+    }
+}

```

Comment:
The class `ShopServiceImpl` should be in the `service.impl` package according to the checklist. Please move it to the correct package to maintain a clear project structure.